### PR TITLE
GLRish for DBDish

### PR DESCRIPTION
**Pull request by salortiz**
Row fetching is lazy so `.allrows()` and `.allrows(:array-of-hash)` methods now returns `Seq`

Named argument `:hash` of `.row` and  is now implemented by role, so drivers can be simplified.
Legacy `.fetchrow` is also implemented by role, so can be removed from drivers.

### CODE REVIEW

**Comment by salortiz** [comment]
@moritz et.al. I would like some opinions before the merger

File: lib/DBIish/CommonTesting.pm6 [hunk 1]
```diff
@@ -333,7 +332,7 @@ method run-tests {
     if $sth.^can('fetchrow_arrayref') {
         ok my $arrayref = $sth.fetchrow_arrayref(), 'called fetchrow_arrayref'; #test 43
         is $arrayref.elems, 4, "fetchrow_arrayref returns 4 fields in a row"; #test 44
-        ok self!magic-cmp($arrayref, [ 'PICO', 'Delish pina colada', '5', 7.90 ]),
+        is $arrayref, [ 'PICO', 'Delish pina colada', '5', 7.90 ],
```
Comment (abraxxa) [inline lib/DBIish/CommonTesting.pm6]: The trailing zero should be removed here as well

**Comment by Skarsnik** [comment]
I don't know if that work well if someone call allrows and disconnect/do another query just behind.

**Comment by salortiz** [comment]
@abraxxa, Indeed, tnks!

**Comment by salortiz** [comment]
@Skarsnik , both cases work well with drivers that uses real prepared transactions (mysql don't, but that should be fixed soon)
The only potentially problematic case is if the user, in the middle of the iteration, re-execute or finish the statement, *and expect* that the current iteration can continue.
In the first case, the iterator is reseted and in the second is terminated. DWIMer IMO.

I'm working on a more robust and orthogonal solution: A new hi-level API that, at statement execution, returns a `DataSet does Positional does Iterable` object to iterate over a `Seq` of `DataRow does Positional does  Associative`. This drivers cleaning process is an important step in that direction.

File: lib/DBIish/CommonTesting.pm6 (hunk 1 above)
Comment (salortiz) [inline lib/DBIish/CommonTesting.pm6]: @abraxxa, Indeed, tnks!

**Comment by moritz** [comment]
:+1: from me.

**Pull request merged**